src/snowflake/connector/cpp/ArrowIterator/Python/Common.hpp

```diff
@@ -35,13 +35,13 @@ class UniqueRef
   {
   }
 
-  UniqueRef(UniqueRef&& other) : UniqueRef(other.detach())
+  explicit UniqueRef(UniqueRef&& other) : UniqueRef(other.release())
   {
   }
 
   UniqueRef& operator=(UniqueRef&& other)
   {
-    m_pyObj = other.detach();
+    reset(other.release());
```

Comment:
@sfc-gh-mkeller  this is what I think a potential bug. Looks like the `m_pyObj` was not properly released by `Py_XDECREF`

Reply:
Ahh, I see. Makes sense. Thank you!